metaflow/plugins/cards/card_modules/components.py

```diff
@@ -477,9 +599,101 @@ class Markdown(UserComponent):
         Text formatted in Markdown.
     """
 
+    REALTIME_UPDATABLE = True
+
+    def update(self, text=None):
+        self._text = text
+
     def __init__(self, text=None):
         self._text = text
 
+    @with_default_component_id
+    @render_safely
+    def render(self):
+        comp = MarkdownComponent(self._text)
+        comp.component_id = self.component_id
+        return comp.render()
+
+
+class ProgressBar(UserComponent):
+    type = "progressBar"
+
+    REALTIME_UPDATABLE = True
+
+    def __init__(self, max=100, label=None, value=0, unit=None, metadata=None):
+        self._label = label
+        self._max = max
+        self._value = value
+        self._unit = unit
+        self._metadata = metadata
+
+    def update(self, new_value, metadata=None):
+        self._value = new_value
+        if metadata is not None:
+            self._metadata = metadata
+
+    @with_default_component_id
+    @render_safely
+    def render(self):
+        data = {
+            "type": self.type,
+            "id": self.component_id,
+            "max": self._max,
+            "value": self._value,
+        }
+        if self._label:
+            data["label"] = self._label
+        if self._unit:
+            data["unit"] = self._unit
+        if self._metadata:
+            data["details"] = self._metadata
+        return data
+
+
+class VegaChart(UserComponent):
+    type = "vegaChart"
+
+    REALTIME_UPDATABLE = True
+
+    def __init__(self, spec, data=None):
+        self._spec = spec
+        self._data = data
+
+    def update(self, data=None, spec=None):
+        if spec is not None:
+            self._spec = spec
+        if data is not None:
+            self._data = data
+
+    @classmethod
+    def from_altair_chart(cls, altair_chart):
```

Comment:
User API signature needs validation : cc @tuulos

Reply:
if I create a chart with `from_altair_chart` initially, do I then update the data with `update(spec=None, data=my_data)` or how would it work?

Reply:
Yes I think you should be able to as long as the data object is the "full object"; The VegaChart object needs the full data object in the data update.